### PR TITLE
Fixed an issue where TA's would be redirected to homepage

### PR DESCRIPTION
**Pull request by Larspolo**
be TA
have can_view_unpublished_assignment
dont have can_edit_assignment
navigate to unpublished assignment -> redirect to homepage

https://sentry.ejournal.app/organizations/ejournal/issues/44

NOTE: the toast can never occur if the assignments are serialized properly (which they are as of now)

---
This change is [Reviewable](https://reviewable.io/reviews/ejourn-al/ejournal/1288)

### CODE REVIEW

**Review by mjvkeulen — APPROVED** [review]

File: src/vue/src/main.js [hunk 1]
```diff
@@ -175,11 +175,21 @@ new Vue({
...
-                        route.name = 'AssignmentEditor'
-                    }
-                } else { // Teacher published route
+                if (assignment.is_published) {
+                    // Published assignments, always go to assignment
+                    route.name = 'Assignment'
+                } else if (!this.$hasPermission('can_view_unpublished_assignment', 'assignment', assignment.id)) {
+                    // Unpublished assignments require the `can_view_unpublished_assignment` permission, else throw
+                    // error and redirect to homepage (redirect is done below)
+                    this.$toasted.error('You are not yet allowed to view this assignment.')
```
Comment (engelhamer) [inline src/vue/src/main.js:184]:
```suggestion
                    this.$toasted.error('You are not allowed to view this assignment.')
```
Could be that this is never even going to be the case.

File: src/vue/src/main.js [hunk 2]
```diff
@@ -175,11 +175,21 @@ new Vue({
...
+                } else if (assignment.is_group_assignment) {
+                    // Group assignments should go to assignment page, even when published
```
Comment (engelhamer) [inline src/vue/src/main.js:186]: This block would never even execute right? It is an `else` to the `is_published` state..?

Comment (Larspolo) [reply src/vue/src/main.js:186]: Unpublished group assignments? Those should be forwarded to the assignment page iso the editor

Comment (engelhamer) [reply src/vue/src/main.js:186]: So then the comment should say unpublished?

**Review by engelhamer — APPROVED** [review]

File: src/vue/src/main.js (hunk 2 above)
Comment (Larspolo) [reply src/vue/src/main.js]: Updated the comment, also better explained the "else"

**Pull request merged**